Ctor 457 new pack network juniper mseries netconf (#4416)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/network-firewalls-juniper-srx-snmp.md

---
id: network-firewalls-juniper-srx-snmp

> title: Juniper SRX

Reviewer: ```suggestion
title: Juniper SRX SNMP
```

Reviewer 2: Non à cause de SAAS-1970

---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Juniper SRX** connector through the

> **Configuration > Monitoring Connectors** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

> **Configuration > Monitoring Connectors** menu:

Reviewer 2: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

| Service Alias     | Service Template                        | Service Description                            |
|:------------------|:----------------------------------------|:-----------------------------------------------|

> | Cpu-Forwarding    | Net-JuniperSRX-Cpu-Forwarding-custom    | Check CPU Usage of packet forwarding engine    |

Reviewer: ```suggestion
| Cpu-Forwarding    | Net-JuniperSRX-Cpu-Forwarding-custom    | Check CPU usage of packet forwarding engine    |
```

> | Cpu-Routing       | Net-JuniperSRX-Cpu-Routing-custom       | Check CPU Usage of routing engine              |

Reviewer: ```suggestion
| Cpu-Routing       | Net-JuniperSRX-Cpu-Routing-custom       | Check CPU usage of routing engine              |
```

> | Cpu-Routing       | Net-JuniperSRX-Cpu-Routing-custom       | Check CPU Usage of routing engine              |

Reviewer 2: ```suggestion
| Cpu-Forwarding    | Net-JuniperSRX-Cpu-Forwarding-custom    | Check CPU usage of packet forwarding engine    |
| Cpu-Routing       | Net-JuniperSRX-Cpu-Routing-custom       | Check CPU usage of routing engine              |
```

> | Flow-Sessions     | Net-JuniperSRX-Flow-Sessions-custom     | Check Packet Forwarding Engine sessions usage  |

Reviewer: ```suggestion
| Flow-Sessions     | Net-JuniperSRX-Flow-Sessions-custom     | Check Packet Forwarding Engine session usage  |
```

> | Flow-Sessions     | Net-JuniperSRX-Flow-Sessions-custom     | Check Packet Forwarding Engine sessions usage  |

Reviewer 2: ```suggestion
| Flow-Sessions     | Net-JuniperSRX-Flow-Sessions-custom     | Check Packet Forwarding Engine session usage  |
```

| Hardware          | Net-JuniperSRX-Hardware-custom          | Check hardware                                 |

> | Memory-Forwarding | Net-JuniperSRX-Memory-Forwarding-custom | Check Memory Usage of packet forwarding engine |

Reviewer: ```suggestion
| Memory-Forwarding | Net-JuniperSRX-Memory-Forwarding-custom | Check memory usage of packet forwarding engine |
```

> | Memory-Routing    | Net-JuniperSRX-Memory-Routing-custom    | Check Memory Usage of routing engine           |

Reviewer: ```suggestion
| Memory-Routing    | Net-JuniperSRX-Memory-Routing-custom    | Check memory usage of routing engine           |
```

> | Memory-Routing    | Net-JuniperSRX-Memory-Routing-custom    | Check Memory Usage of routing engine           |

Reviewer 2: ```suggestion
| Memory-Forwarding | Net-JuniperSRX-Memory-Forwarding-custom | Check Memory usage of packet forwarding engine |
| Memory-Routing    | Net-JuniperSRX-Memory-Routing-custom    | Check Memory usage of routing engine           |
```

> The services listed above are created automatically when the **Net-Juniper-SRX-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias        | Service Template                           | Service Description                                                                                                                       | Discovery  |
|:---------------------|:-------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------|:----------:|
| Cp-Sessions          | Net-JuniperSRX-Cp-Sessions-custom          | Check CP ('central point') sessions usage                                                                                                 |            |

> | Disk-Generic-Id      | Net-JuniperSRX-Disk-Generic-Id-custom      | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free space remaining                         |            |

Reviewer: ```suggestion
| Disk-Generic-Id      | Net-JuniperSRX-Disk-Generic-Id-custom      | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free remaining                          space |            |
```

> | Disk-Generic-Id      | Net-JuniperSRX-Disk-Generic-Id-custom      | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free space remaining                         |            |

Reviewer 2: ```suggestion
| Disk-Generic-Id      | Net-JuniperSRX-Disk-Generic-Id-custom      | Check the rate of free space on the disk (use the ID). Thresholds can be in percentage or in free remaining space |            |
```

| Disk-Generic-Name    | Net-JuniperSRX-Disk-Generic-Name-custom    | Check the rate of free space on the disk (use the Name. Difficult to use                                                                  |            |

> | Disk-Global          | Net-JuniperSRX-Disk-Global-custom          | Check the rate of free space on the disk (use the Name. Difficult to use                                                                  | X          |

Reviewer: Difficult to use?

> | Disk-Global          | Net-JuniperSRX-Disk-Global-custom          | Check the rate of free space on the disk (use the Name. Difficult to use                                                                  | X          |

Reviewer 2: ```suggestion
| Disk-Generic-Name    | Net-JuniperSRX-Disk-Generic-Name-custom    | Check the rate of free space on the disk (use the Name.                                                                  |            |
| Disk-Global          | Net-JuniperSRX-Disk-Global-custom          | Check the rate of free space on the disk (use the Name.                                                                  | X          |
```

<Tabs groupId="sync">
<TabItem value="Cp-Sessions" label="Cp-Sessions">

| Nom | Unité |

> |:----|:------|

Reviewer 2: ```suggestion
| Name                 | Unit  |
|:---------------------|:------|
| sessions\_*instance* | count |
```

</TabItem>
<TabItem value="Cpu-Forwarding" label="Cpu-Forwarding">

| Nom | Unité |

> |:----|:------|

Reviewer 2: ```suggestion
| Name            | Unit  |
|:----------------|:------|
| cpu\_*instance* | %     |
```

</TabItem>
<TabItem value="Flow-Sessions" label="Flow-Sessions">

| Nom | Unité |

> |:----|:------|

Reviewer 2: ```suggestion
| Name                 | Unit  |
|:---------------------|:------|
| sessions\_*instance* | count |
```

</TabItem>
<TabItem value="Hardware" label="Hardware">

| Name                     | Unit  |
|:-------------------------|:------|
| hardware.fru.count       | count |
| fru status               | N/A   |
| hardware.operating.count | count |
| operating status         | N/A   |
| hardware.alarm.count     | count |
| alarm status             | N/A   |

</TabItem>
<TabItem value="Memory-Forwarding" label="Memory-Forwarding">

| Nom | Unité |

> |:----|:------|

Reviewer 2: ```suggestion
| Name | Unit |
|:----------------|:------|
| mem\_*instance* | %     |
```

</TabItem>
<TabItem value="Memory-Routing" label="Memory-Routing">

| Name                             | Unit  |
|:---------------------------------|:------|
| *memory*#memory.usage.bytes      | B     |
| *memory*#memory.free.bytes       | B     |
| *memory*#memory.usage.percentage | %     |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Traffic-*" label="Traffic-*">

> Coming soon

Reviewer 2: ```suggestion
| Name                                         | Unit  |
|:-----------------------------------------------------|:------|
| *interface_name*#status                              | N/A   |
| *interface_name*#interface.traffic.in.bitspersecond  | b/s   |
| *interface_name*#interface.traffic.out.bitspersecond | b/s   |

> Applies to the following service templates: Traffic-Generic-Id, Traffic-Generic-Name, Traffic-Global
```

### Pack

The installation procedures for monitoring connectors are slightly different depending on [whether your license is offline or online](../getting-started/how-to-guides/connectors-licenses.md).

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connectors** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

> **Configuration > Monitoring Connectors** menu.

Reviewer 2: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Juniper SRX** connector through

> the **Configuration > Monitoring Connectors** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

> the **Configuration > Monitoring Connectors** menu.

Reviewer 2: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

</TabItem>
<TabItem value="Cpu-Routing" label="Cpu-Routing">

| Macro           | Description                                                                                        | Default value     | Mandatory   |
|:----------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER          | Filter operating (default: <'routing\|fpc'\>)                                                    | routing           |             |

Reviewer 2: ```suggestion
| FILTER          | Filter operating                                                  | routing           |             |
```

> | FILTER          | Filter operating (default: C\<'routing\|fpc'\>)                                                    | routing           |             |

Reviewer: C?

> | FILTER          | Filter operating (default: C\<'routing\|fpc'\>)                                                    | routing           |             |

Reviewer 2: ```suggestion
| FILTER          | Filter operating (default: <'routing\|fpc'\>)                                                    | routing           |             |
```

</TabItem>
<TabItem value="Memory-Routing" label="Memory-Routing">

| Macro             | Description                                                                                        | Default value     | Mandatory   |
|:------------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER            | Filter operating (default: 'routing\|fpc')                                                         | routing           |             |

Reviewer 2: ```suggestion
| FILTER            | Filter operating                                                       | routing           |             |
```

</TabItem>
<TabItem value="Memory-Routing" label="Memory-Routing">

| Macro             | Description                                                                                        | Default value     | Mandatory   |
|:------------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER            | Filter operating                                                       | routing           |             |

Reviewer: filter operating???

| CRITICALIN     | Set critical threshold for in traffic                                                                                                                                                                                                                    | 90                |             |
| WARNINGOUT     | Set warning threshold for out traffic                                                                                                                                                                                                                    | 80                |             |
| CRITICALOUT    | Set critical threshold for out traffic                                                                                                                                                                                                                   | 90                |             |

> | CRITICALSTATUS | Define the conditions to match for the status to be CRITICAL (default: C\<'%\{admstatus\} eq "up" and %\{opstatus\} ne "up"'\>). You can use the following variables: C\<%\{admstatus\}\>, C\<%\{opstatus\}\>, C\<%\{duplexstatus\}\>, C\<%\{display\}\> |                   |             |

Reviewer: C?

> | CRITICALSTATUS | Define the conditions to match for the status to be CRITICAL (default: C\<'%\{admstatus\} eq "up" and %\{opstatus\} ne "up"'\>). You can use the following variables: C\<%\{admstatus\}\>, C\<%\{opstatus\}\>, C\<%\{duplexstatus\}\>, C\<%\{display\}\> |                   |             |

Reviewer 2: ```suggestion
| CRITICALSTATUS | Define the conditions to match for the status to be CRITICAL (default: '%\{admstatus\} eq "up" and %\{opstatus\} ne "up"'). You can use the following variables: %\{admstatus\}, %\{opstatus\}, %\{duplexstatus\}, %\{display\} |                   |             |
```

```bash
/usr/lib/centreon/plugins/centreon_juniper_srx.pl \
	--plugin=network::juniper::srx::snmp::plugin \
	--mode=memory \
	--hostname=10.0.0.1 \
	--snmp-version='2c' \
	--snmp-community='my-snmp-community'  \
	--filter='routing' \
	--warning-usage='' \
	--critical-usage='' \
	--warning-usage-free='' \
	--critical-usage-free='' \
	--warning-usage-prct='80' \
	--critical-usage-prct='90' \
	--verbose
```

The expected command output is shown below:

```bash

> OK: All memories are ok | 'memory1#memory.usage.bytes'=40460B;;;0;total 'memory2#memory.usage.bytes'=15823B;;;0;total 'memory1#memory.free.bytes'=79777B;;;0;total 'memory2#memory.free.bytes'=26745B;;;0;total 'memory1#memory.usage.percentage'=38436%;;;0;100 'memory2#memory.usage.percentage'=27676%;;;0;100 

Reviewer: thresholds in output?

> OK: All memories are ok | 'memory1#memory.usage.bytes'=40460B;;;0;total 'memory2#memory.usage.bytes'=15823B;;;0;total 'memory1#memory.free.bytes'=79777B;;;0;total 'memory2#memory.free.bytes'=26745B;;;0;total 'memory1#memory.usage.percentage'=38436%;;;0;100 'memory2#memory.usage.percentage'=27676%;;;0;100 

Reviewer 2: ```suggestion
OK: All memories are ok | 'memory1#memory.usage.bytes'=40460B;;;0;total 'memory2#memory.usage.bytes'=15823B;;;0;total 'memory1#memory.free.bytes'=79777B;;;0;total 'memory2#memory.free.bytes'=26745B;;;0;total 'memory1#memory.usage.percentage'=38%;80;90;0;100 'memory2#memory.usage.percentage'=27%;80;90;0;100 
```
